### ISSUE TITLE
Investigate implementing SIMD based binary comparison ops

### ISSUE THREAD

**Issue by romeric** [romeric/Fastor#49]
We lack SIMD based binary comparison operators. At the moment binary_cmp_ops have a pretty decent speed. Compilers are very good at generating binary comparison operators using SSE/AVX/AVX2 from scalar code though. 

The following AVX code is slightly/marginally faster than our current implementation but still nearly twice slower than a plain scalar based implementation (that the compiler can optimise)  

~~~c++
template<>
FASTOR_INLINE SIMDVector<bool,SIMDVector<double,256>::Size*sizeof(bool)*8> 
operator>(const SIMDVector<double,256> &a, const SIMDVector<double,256> &b) {
    constexpr FASTOR_INDEX Size = SIMDVector<double,256>::Size;
    __m256d cmp = _mm256_cmp_pd(a.value,b.value,_CMP_GT_OQ);
    int bitmask = _mm256_movemask_pd(cmp);
    SIMDVector<bool,Size*sizeof(bool)*8> out;
    bool FASTOR_ALIGN val_out[Size];
    out.store(val_out);
    for (FASTOR_INDEX i=0; i<Size; ++i) {
        val_out[i] = (bitmask & (1<<i)) != 0;
    }
    out.load(val_out);
    return out;
}

~~~

**Comment by romeric** [comment]
The best solution at the moment seems like allowing the compiler to emit the right code by dispatching the construction of tensor to scalar code i.e. using `eval_s<T>` only.

**Comment by romeric** [comment]
fixed in 71af418. Closing for now.
